app/features/contacts/contact-repository.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  type BoardwalkDb,
+  type BoardwalkDbContact,
+  boardwalkDb,
+} from '../boardwalk-db/database';
+import type { UserProfile } from '../user/user';
+import type { Contact } from './contact';
+
+type CreateContact = {
+  /** Id of the user creating the contact */
+  ownerId: string;
+  /** Username of the user within this app to add as a contact */
+  username: string;
+};
+
+export class ContactRepository {
+  constructor(private readonly db: BoardwalkDb) {}
+
+  async get(contactId: string) {
+    const query = this.db.from('contacts').select().eq('id', contactId);
+
+    const { data, error } = await query.single();
+
+    if (error) {
+      throw new Error('Failed to get contact', error);
+    }
+
+    return ContactRepository.toContact(data);
+  }
+
+  /**
+   * Gets all contacts for a user
+   * @param ownerId - The ID of the user whose contacts to fetch
+   * @returns Array of contacts
+   */
+  async getAll(
+    ownerId: string,
+    options?: { abortSignal?: AbortSignal },
+  ): Promise<Contact[]> {
+    const query = this.db
+      .from('contacts')
+      .select()
+      .eq('owner_id', ownerId)
+      .order('username', { ascending: true }); // sort alphabetically
```

Comment:
we should probably add a limit to this query so it doesn't try to pull too much. e.g. limit 100/200 or something like that